### PR TITLE
[studio-ui] Edit Panel Fixes & Parsing Repeats content type fields

### PR DESCRIPTION
**Pull request by aachavarria**

### CODE REVIEW

File: ui/app/src/components/PreviewAddressBar/PreviewAddressBar.tsx [hunk 1]
```diff
@@ -160,9 +160,11 @@ export function PreviewAddressBar(props: AddressBarProps) {
...
       <PreviewForwardButton />
       <Tooltip title={<FormattedMessage id="previewAddressBar.reloadButtonLabel" defaultMessage="Reload this page" />}>
-        <IconButton onClick={onRefresh} size="large" disabled={disabled}>
-          <RefreshRounded />
-        </IconButton>
+        <div>
+          <IconButton onClick={onRefresh} size="large" disabled={disabled}>
+            <RefreshRounded />
+          </IconButton>
+        </div>
```
Comment (rart) [inline ui/app/src/components/PreviewAddressBar/PreviewAddressBar.tsx]:
```suggestion
        <span>
          <IconButton onClick={onRefresh} size="large" disabled={disabled}>
            <RefreshRounded />
          </IconButton>
        </span>
```

File: ui/app/src/utils/model.ts [hunk 2]
```diff
@@ -14,10 +14,13 @@
...
  */
 
-import { nou, retrieveProperty, setProperty } from './object';
+import { notNullOrUndefined, nou, retrieveProperty, setProperty } from './object';
 import { removeLastPiece } from './string';
 import ContentInstance from '../models/ContentInstance';
 import LookupTable from '../models/LookupTable';
+import { ModelHierarchyMap } from './content';
+import { forEach } from './array';
+import * as Model from '@craftercms/studio-guest/build_tsc/utils/model';
```
Comment (rart) [inline ui/app/src/utils/model.ts]: use local path based import

Comment (aachavarria) [reply ui/app/src/utils/model.ts]: I don't get it

File: ui/guest/src/store/reducers/root.ts [hunk 3]
```diff
@@ -129,6 +129,7 @@ const reducer = createReducer(initialState, {
         return { ...state, highlighted: { [record.id]: highlight } };
       } else if (highlightMode === HighlightMode.MOVE_TARGETS) {
         const iceId = record.iceIds[0];
+
```
Comment (rart) [inline ui/guest/src/store/reducers/root.ts]:
```suggestion
```

File: ui/guest/src/classes/ICERegistry.ts [hunk 4]
```diff
@@ -356,20 +359,30 @@ export function isMovableType(id: number): boolean {
...
-    // - Field of a page
-    return exists({
-      modelId: modelHierarchyMap[modelId].parentId,
-      fieldId: modelHierarchyMap[modelId].parentContainerFieldPath,
-      index: modelHierarchyMap[modelId].parentContainerFieldIndex
-    });
+    if (isSimple(fieldId)) {
+      // Can be...
+      // - Field of a component (possible move target)
+      // - Field of a repeat (certain move target)
```
Comment (rart) [inline ui/guest/src/classes/ICERegistry.ts]:
```suggestion
```

File: ui/guest/src/react/GuestProxy.tsx [hunk 5]
```diff
@@ -102,18 +102,45 @@ export function GuestProxy() {
       // Because more than one item may change their index, we need to deregister all items before
       // proceeding to registering again. Registering immediately after de-registering on the same
       // loop causes misjudging ICE records refCounts due to displaced collection items.
+
+      function updateRepeatGroupsItems(element, fieldId, index, newIndex, array) {
```
Comment (rart) [inline ui/guest/src/react/GuestProxy.tsx]: `array` needs a better name

File: ui/guest/src/classes/ICERegistry.ts [hunk 6]
```diff
@@ -78,6 +78,9 @@ let rid = 0;
 /* private */
 const registry: Map<number, ICERecord> = new Map();
 
+// @ts-ignore
+window.registry = registry;
+
```
Comment (rart) [inline ui/guest/src/classes/ICERegistry.ts]: x

File: ui/guest/src/classes/ContentController.ts [hunk 7]
```diff
@@ -690,12 +721,12 @@ export function deleteItem(modelId: string, fieldId: string, index: number | str
...
-    fieldId,
-    index,
-    parentModelId: getParentModelId(modelId, models, modelHierarchyMap)
-  });
+  // post(deleteItemOperation.type, {
+  //   modelId,
+  //   fieldId,
+  //   index,
+  //   parentModelId: getParentModelId(modelId, models, modelHierarchyMap)
+  // });
```
Comment (rart) [inline ui/guest/src/classes/ContentController.ts]: 👆

**Review by rart — COMMENTED** [review]
Let's review together.

File: ui/app/src/utils/model.ts (hunk 2 above)
Comment (aachavarria) [reply ui/app/src/utils/model.ts]: pending

File: ui/guest/src/classes/ContentController.ts [hunk 8]
```diff
@@ -671,11 +717,32 @@ export function deleteItem(modelId: string, fieldId: string, index: number | str
...
+
   // Deleting item from modelHierarchyMap
-  if (typeof result[0] === 'string') {
-    delete modelHierarchyMap[item];
+  if (typeof collection[0] === 'string') {
+    deleteItemFromHierarchyMap(item as string);
+    updateHierarchyMapIndexes(result);
   } else {
-    // TODO: check modelHierarchyMap[item].children
+    const modelsToDeleted = [];
```
Comment (rart) [inline ui/guest/src/classes/ContentController.ts]:
```suggestion
    const modelsToDelete = [];
```

**Comment by rart** [comment]
@aachavarria please pull and fix conflicts

**Review by rart — APPROVED** [review]

**Pull request merged**